pkg/webhook/admission/pod/storage_initializer_injector.go

```
@@ -237,6 +251,13 @@ func (mi *StorageInitializerInjector) InjectStorageInitializer(pod *corev1.Pod)
 	transformerContainer := getContainerWithName(pod, constants.TransformerContainerName)
 	workerContainer := getContainerWithName(pod, constants.WorkerContainerName)
 
+	if enabledRemoteEnv {
+		if _, exists := utils.GetEnvVarValue(userContainer.Env, RemoteStorageEnvVarName); exists {
+			return fmt.Errorf("container %s already defines %s", userContainer.Name, RemoteStorageEnvVarName)
```

The webhook can be called multiple times so we need to make it idempotent, we do not need to return an error instead check if the environment does not exist we inject it.

---

Good point, I’ll update it to make the injection idempotent as suggested. Thanks!